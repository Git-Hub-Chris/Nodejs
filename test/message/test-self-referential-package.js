'use strict';

require('../common');
const { spawn } = require('child_process');

const fixtures = require('../common/fixtures');
const selfRefModule = fixtures.path('self_ref_module_with_colon');

spawn(process.argv0, [`${selfRefModule}/bin.js`], {
  stdio: 'inherit'
});
