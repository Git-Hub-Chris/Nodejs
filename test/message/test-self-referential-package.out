TypeError [ERR_INVALID_MODULE_SPECIFIER]: Invalid module "file:package" is not a valid package name imported from *
    at new NodeError (node:internal/errors:*)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:*)
    at Function.Module._load (node:internal/modules/cjs/loader:*)
    at Module.require (node:internal/modules/cjs/loader:*)
    at require (node:internal/modules/cjs/helpers:*)
    at Object.<anonymous> (*)
    at Module._compile (node:internal/modules/cjs/loader:*)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:*)
    at Module.load (node:internal/modules/cjs/loader:*)
    at Function.Module._load (node:internal/modules/cjs/loader:*) {
  code: 'ERR_INVALID_MODULE_SPECIFIER'
}
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/package' imported from *
    at new NodeError (node:internal/errors:*)
    at finalizeResolution (node:internal/modules/esm/resolve:*)
    at moduleResolve (node:internal/modules/esm/resolve:*)
    at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:*)
    at Loader.resolve (node:internal/modules/esm/loader:*)
    at Loader.getModuleJob (node:internal/modules/esm/loader:*)
    at Loader.import (node:internal/modules/esm/loader:*)
    at importModuleDynamically (node:internal/modules/cjs/loader:*)
    at importModuleDynamicallyWrapper (node:internal/vm/module:*)
    at importModuleDynamically (node:vm:*) {
  code: 'ERR_MODULE_NOT_FOUND'
}
TypeError [ERR_INVALID_MODULE_SPECIFIER]: Invalid module "file%%3Apackage" is not a valid package name imported from *
    at new NodeError (node:internal/errors:*)
    at parsePackageName (node:internal/modules/esm/resolve:*)
    at packageResolve (node:internal/modules/esm/resolve:*)
    at moduleResolve (node:internal/modules/esm/resolve:*)
    at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:*)
    at Loader.resolve (node:internal/modules/esm/loader:*)
    at Loader.getModuleJob (node:internal/modules/esm/loader:*)
    at Loader.import (node:internal/modules/esm/loader:*)
    at importModuleDynamically (node:internal/modules/cjs/loader:*)
    at importModuleDynamicallyWrapper (node:internal/vm/module:*) {
  code: 'ERR_INVALID_MODULE_SPECIFIER'
}
